pipes/fetch-job-scripts.pipe.js

```diff
@@ -91,10 +91,11 @@ const getJobScripts = async (context, job) => {
       scriptContentsAfterCut = scriptContents;
 
       if (context.config.regex) {
-        const regex = new RegExp(context.config.regex, 'g');
+        const regex = new RegExp(context.config.regex, context.config.regexOptions);
         const regexResult = regex.exec(scriptContents);
+        // Retrieve the full strings of chars matched.
         if (regexResult && regexResult.length > 1) {
-          [, scriptContents] = regexResult;
+          scriptContents = regexResult[0];
```

Comment:
afaik it will only return the first result, if I understand correctly, what you want to achieve is a concatenated string of all the matched results, so you probably want to do this: https://stackoverflow.com/questions/6323417/how-do-i-retrieve-all-matches-for-a-regular-expression-in-javascript

correct my if i'm wrong

Reply:
Hmmm. I used the example in the official documentation : https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec#Description

Let me know if you still think this needs revision

Reply:
Is there any update regarding this?